### PR TITLE
Chore/exp

### PR DESCRIPTION
**Pull request by DavidMinarsch**
## Proposed changes

Experimental branch

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: autonomy/replay/utils.py [hunk 1]
```diff
@@ -21,12 +21,14 @@
...
 import json
 from pathlib import Path
+from typing import Any, Dict
 
 from autonomy.deploy.constants import PERSISTENT_DATA_DIR, TM_STATE_DIR
 
 
 def fix_address_books(build_dir: Path) -> None:
     """Update address books in data dump to use them in replays."""
+    # TODO: add json validation and error handling
```
Comment (DavidMinarsch) [inline autonomy/replay/utils.py:31]: This is a big issue throughout. Various config files have no validation and then also no error handling when loading/accessing

File: autonomy/replay/agent.py [hunk 2]
```diff
@@ -67,6 +71,7 @@ def start(
         agent_dir = Path(self.agent_dir.name)
         os.chdir(agent_dir)
 
+        # TODO: replace with more appropriate env var name; we mean an agent here
```
Comment (DavidMinarsch) [inline autonomy/replay/agent.py:74]: what is VALORY_APPLICATION, an agent?

File: autonomy/deploy/generators/docker_compose/base.py [hunk 3]
```diff
@@ -103,7 +104,7 @@ def build_agent_config(  # pylint: disable=too-many-arguments
 class DockerComposeGenerator(BaseDeploymentGenerator):
     """Class to automate the generation of Deployments."""
 
-    output_name: str = "docker-compose.yaml"
+    output_name: str = DOCKER_COMPOSE_YAML
     deployment_type: str = "docker-compose"
```
Comment (DavidMinarsch) [inline autonomy/deploy/generators/docker_compose/base.py:108]: also a constant

File: autonomy/deploy/constants.py [hunk 4]
```diff
@@ -50,7 +50,6 @@
 KEY_SCHEMA_PRIVATE_KEY = "private_key"
 
 
-DEFAULT_ABCI_BUILD_DIR = "abci_build"
```
Comment (DavidMinarsch) [inline autonomy/deploy/constants.py:53]: already exists elsewhere in `autonomy/constants.py`

File: autonomy/deploy/base.py [hunk 5]
```diff
@@ -54,6 +54,7 @@
 }
 
 
+# TODO: how dpes this relate to `autonomy/services`? Unify
```
Comment (DavidMinarsch) [inline autonomy/deploy/base.py:57]: I have no idea why we have this and the Service config...

File: autonomy/constants.py [hunk 6]
```diff
@@ -39,6 +42,13 @@
 HARDHAT_IMAGE_NAME = os.environ.get(
     "HARDHAT_IMAGE_NAME", "valory/open-autonomy-hardhat"
 )
+SERVICE_REGISTRY_IMAGE_NAME = os.environ.get(
```
Comment (DavidMinarsch) [inline autonomy/constants.py:45]: adjusted to above format

File: autonomy/cli/publish.py [hunk 7]
```diff
@@ -130,6 +133,7 @@ def publish_service_local(ctx: Context, public_id: PublicId) -> None:
     author_dir = Path(target_dir).parent
     if not os.path.exists(author_dir):
         os.makedirs(author_dir, exist_ok=True)
+    # TODO: also make services dir?
```
Comment (DavidMinarsch) [inline autonomy/cli/publish.py:136]: not clear why we need to make an author dir, but not the services dir

File: autonomy/cli/packages.py [hunk 8]
```diff
@@ -38,13 +38,12 @@
...
 @pass_ctx
 def lock_packages(ctx: Context, check: bool) -> None:
-    """Lock packages"""
+    """Lock local packages."""
 
     packages_dir = Path(ctx.registry_path)
 
     try:
         if check:
-            packages_dir = Path(ctx.registry_path)
```
Comment (DavidMinarsch) [inline autonomy/cli/packages.py:47]: already above

File: autonomy/analyse/abci/app_spec.py [hunk 9]
```diff
@@ -430,10 +435,14 @@ def check_all(cls, packages_dir: Path) -> None:
         did_not_match = []
         fsm_specifications = sorted(
             [
-                *packages_dir.glob("**/fsm_specification.yaml"),
-                *packages_dir.glob("**/fsm_specification_composition.yaml"),
+                *packages_dir.glob(f"**/{DEFAULT_FSM_SPEC_YAML}"),
             ]
         )
+        # TODO: fix this implementation:
```
Comment (DavidMinarsch) [inline autonomy/analyse/abci/app_spec.py:441]: I think this is the reason for the flaky tests

File: autonomy/analyse/abci/app_spec.py [hunk 10]
```diff
@@ -430,10 +435,14 @@ def check_all(cls, packages_dir: Path) -> None:
         did_not_match = []
         fsm_specifications = sorted(
             [
-                *packages_dir.glob("**/fsm_specification.yaml"),
-                *packages_dir.glob("**/fsm_specification_composition.yaml"),
```
Comment (DavidMinarsch) [inline autonomy/analyse/abci/app_spec.py:434]: There's no reason for two formats. We originally had it when the design was less progressed.

File: autonomy/cli/analyse.py [hunk 11]
```diff
@@ -58,6 +61,7 @@ def generate_abci_app_specs(
 
     module_name, class_name = app_class.rsplit(".", 1)
 
+    # TODO: extract into helper
```
Comment (angrybayblade) [inline autonomy/cli/analyse.py:63]: @DavidMinarsch Can you provide more context to this?

File: autonomy/cli/deploy.py [hunk 12]
```diff
@@ -317,10 +316,11 @@ def run_deployment_from_token(  # pylint: disable=too-many-arguments, too-many-l
     run_deployment(build_dir)
 
 
+# TODO: extract into appropriate utils with validations
```
Comment (angrybayblade) [inline autonomy/cli/deploy.py:319]: https://github.com/valory-xyz/open-autonomy/issues/1453

File: autonomy/cli/fetch.py [hunk 13]
```diff
@@ -91,6 +91,7 @@ def fetch(
         raise click.ClickException(str(e)) from e
 
 
+# TODO: extract below into appropriate UTILS
```
Comment (angrybayblade) [inline autonomy/cli/fetch.py:94]: This will be done when we perform refactors

File: autonomy/cli/hash.py [hunk 14]
```diff
@@ -45,6 +45,7 @@
 from autonomy.configurations.base import PACKAGE_TYPE_TO_CONFIG_CLASS
 
 
+# TODO: extract into utils; probably a duplication?
```
Comment (angrybayblade) [inline autonomy/cli/hash.py:48]: Same as above

File: autonomy/cli/hash.py [hunk 15]
```diff
@@ -64,6 +65,8 @@ def load_configuration(
     return cast(PackageConfiguration, configuration_obj)
 
 
+# TODO: extract into utils
+# Add input validations
```
Comment (angrybayblade) [inline autonomy/cli/hash.py:69]: Input validation for `packages.json`?

File: autonomy/cli/publish.py (hunk 7 above)
Comment (angrybayblade) [reply autonomy/cli/publish.py:136]: `os.makedirs` work recursively and creates missing parent directories as well

Comment (DavidMinarsch) [reply autonomy/cli/publish.py:136]: So then why are we creating the one for the author?

**Pull request merged**